carapace-server/src/main/java/org/carapaceproxy/api/CertificatesResource.java

```diff
@@ -220,22 +220,19 @@ public Response downloadCertificateById(@PathParam("certId") String certId) thro
 
     private CertificateBean findCertificateById(String certId) {
         HttpProxyServer server = (HttpProxyServer) context.getAttribute("server");
-        RuntimeServerConfiguration conf = server.getCurrentConfiguration();
-        Map<String, SSLCertificateConfiguration> certificateList = conf.getCertificates();
-        if (certificateList.containsKey(certId)) {
-            SSLCertificateConfiguration certificate = certificateList.get(certId);
+        SSLCertificateConfiguration certificate = server.getCurrentConfiguration().getCertificates().get(certId);
+        if (certificate != null) {
             CertificateBean certBean = new CertificateBean(
                     certificate.getId(),
                     certificate.getHostname(),
                     certificateModeToString(certificate.getMode()),
                     certificate.isDynamic(),
                     certificate.getFile()
             );
-
             if (certificate.isDynamic()) {
                 certBean.setStatus(certificateStateToString(null)); // unknown
                 try {
-                    CertificateData cert = server.getDynamicCertificatesManager().getCertificateDataForDomain(certificate.getHostname());
+                    CertificateData cert = server.getDynamicCertificatesManager().getCertificateDataForDomain(certificate.getId());
```

Comment:
is this a breaking change ?
is it compatible with existing data ?

Reply:
All tests passes, hostname == id whether not wildcard. Existing wildcard certificates are all in "file-mode", so none is currently managed by certificatemanager